Sources/Instrumentation/MetricKit/MetricKitInstrumentation.swift

```diff
@@ -0,0 +1,608 @@
+                            )
+                        }
+                    }
+                }
+                span.end()
+            }
+        }
+    }
+
+    @available(iOS 14.0, macOS 12.0, macCatalyst 14.0, visionOS 1.0, *)
+    public func reportDiagnostics(payload: MXDiagnosticPayload) {
+        let span = getMetricKitTracer().spanBuilder(spanName: "MXDiagnosticPayload")
+            .setStartTime(time: payload.timeStampBegin)
+            .startSpan()
+        defer { span.end() }
+
+        let logger = OpenTelemetry.instance.loggerProvider.get(
+            instrumentationScopeName: metricKitInstrumentationName
+        )
+
+        let now = Date()
+
+        // A helper for looping over the items in an optional list and logging each one.
+        func logForEach<T>(
+            _ parent: [T]?,
+            _ namespace: String,
+            using closure: (T) -> ([String: AttributeValueConvertable], [String: AttributeValueConvertable])
+        ) {
+            if let arr = parent {
+                for item in arr {
+                    var attributes: [String: AttributeValue] = [
+                        "name": "metrickit.diagnostic.\(namespace)".attributeValue()
+                    ]
+                    let (namespacedAttrs, globalAttrs) = closure(item)
+
+                    // Add namespaced attributes with prefix
+                    for (key, value) in namespacedAttrs {
+                        let namespacedKey = "metrickit.diagnostic.\(namespace).\(key)"
+                        attributes[namespacedKey] = value.attributeValue()
+                    }
+
+                    // Add global attributes without prefix (for standard OTel attributes)
+                    for (key, value) in globalAttrs {
+                        attributes[key] = value.attributeValue()
+                    }
+
+                    logger.logRecordBuilder()
+                        .setTimestamp(payload.timeStampEnd)
+                        .setObservedTimestamp(now)
+                        .setAttributes(attributes)
+                        .emit()
+                }
+            }
+        }
+
+        #if !os(macOS)
+        if #available(iOS 16.0, *) {
+            logForEach(payload.appLaunchDiagnostics, "app_launch") {
+                (["launch_duration": $0.launchDuration], [:])
+            }
+        }
+        #endif
+
+        logForEach(payload.diskWriteExceptionDiagnostics, "disk_write_exception") {
+            (["total_writes_caused": $0.totalWritesCaused], [:])
+        }
+        logForEach(payload.hangDiagnostics, "hang") {
+            let callStackTree = $0.callStackTree
+            let stacktraceJson = String(
+                decoding: callStackTree.jsonRepresentation(),
+                as: UTF8.self
+            )
+
+            let namespacedAttrs: [String: AttributeValueConvertable] = [
+                "hang_duration": $0.hangDuration,
+                "exception.stacktrace_json": stacktraceJson
```

Comment:
Can we just keep the [exception.stacktrace](https://opentelemetry.io/docs/specs/semconv/registry/attributes/exception/)` semconv? I think the different stacktrace representations are already implied, and if not then we can add MetricKit json format to the "Stacktrace representation" list